src/component.cpp

```diff
@@ -142,12 +142,12 @@ std::string Component::doSerialisation(FORMATS format) const
         std::string componentName = getName();
         if (componentName.length()) {
             repr += " name=\"" + componentName + "\"";
-            if (variableCount() > 0) {
-                endTag = true;
-                repr += ">";
-                for(size_t i = 0; i < variableCount(); i++) {
-                    repr += getVariable(i)->serialise(format);
-                }
+        }
+        if (variableCount() > 0) {
+            endTag = true;
+            repr += ">";
+            for(size_t i = 0; i < variableCount(); i++) {
```

Comment:
Re: "++i" instead of "i++", the former does indeed seem to be the preferred form in the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) (GCSG) we are using as a guideline. The same goes for the spacing in "`for (`". I can go through and do a formatting fix for instances of this and other things we pick up throughout the code after this bugfix and the initialValues pulls go through.

As for 

`for (size_t i = 0, iMax = variableCount(); i < iMax; ++i) {`

versus

`for (size_t i = 0; i < variableCount(); ++i) {`

I personally prefer the latter for succinctness but don't see any hard and fast rules in the GCSG. They do seem to give some examples [more in line with the latter](https://google.github.io/styleguide/cppguide.html#Implementation_Comments) e.g., 

`for (int i = 0; i < result->size(); i++) {`

But I guess consistency is what really matters- I could be convinced if more prefer the "`iMax`" style. Any opinions @nickerso , @hsorby , or @jonc125 ?